Include the icon in Anaconda's keyboard selection screen.
https://bugzilla.redhat.com/show_bug.cgi?id=469165

Lubomir Rintel <lkundrak@v3.sk>

diff -up system-config-keyboard-1.2.15/src/keyboard_gui.py.icon system-config-keyboard-1.2.15/src/keyboard_gui.py
--- system-config-keyboard-1.2.15/src/keyboard_gui.py.icon	2008-11-12 23:42:29.000000000 +0100
+++ system-config-keyboard-1.2.15/src/keyboard_gui.py	2008-11-12 23:47:30.000000000 +0100
@@ -6,9 +6,10 @@
 ## Jeremy Katz <katzj@redhat.com>
 ## Chris Lumens <clumens@redhat.com>
 ## Bill Nottingham <notting@redhat.com>
-## Lubomir Kundrak <lkundrak@redhat.com>
+## Lubomir Rintel <lkundrak@v3.sk>
 ##
 ## Copyright (C) 2002, 2003, 2007, 2008 Red Hat, Inc.
+## Copyright (C) 2008 Lubomir Rintel
 ##
 ## This program is free software; you can redistribute it and/or modify
 ## it under the terms of the GNU General Public License as published by
@@ -136,14 +137,21 @@ class moduleClass(Module):
 
         self.kbd = kbd
 
-        self.vbox = gtk.VBox()
-        self.vbox.set_spacing(10)
-        self.vbox.set_border_width(10)
-
-        label = gtk.Label(_("Select the appropriate keyboard for the system."))
-        label.set_alignment(0.0, 0.5)
-        label.set_size_request(500, -1)
-        self.vbox.pack_start(label, False)
+        self.vbox = gtk.VBox(False, 10)
+
+        iconBox = gtk.HBox(False, 5)
+        iconBox.pack_start(loadToImage(iconFile))
+
+        msgLabel = gtk.Label(_("Select the appropriate keyboard for the system."))
+        msgLabel.set_line_wrap (True)
+        msgLabel.set_size_request(250, -1)
+
+        iconBox.pack_start(msgLabel)
+        iconBox.set_border_width(5)
+
+        align = gtk.Alignment()
+        align.add(iconBox);
+        self.vbox.pack_start(align, False)
 
         if self.kbd.beenset:
             default = defaultByLang
@@ -192,6 +200,7 @@ class moduleClass(Module):
         self.modelViewSW.set_policy(gtk.POLICY_AUTOMATIC, gtk.POLICY_AUTOMATIC)
         self.modelViewSW.set_shadow_type(gtk.SHADOW_IN)
         self.modelViewSW.add(self.modelView)
+        self.modelViewSW.set_border_width(5)
 
         self.vbox.pack_start(self.modelViewSW, True)
 
@@ -243,7 +252,6 @@ class moduleClass(Module):
         self.mainWindow.connect("destroy", self.destroy)
         self.mainWindow.set_border_width(10)
         self.mainWindow.set_size_request(400, 350)
-        self.mainWindow.set_position(gtk.WIN_POS_CENTER)
 
         self.mainWindow.set_icon(iconPixbuf)
         self.mainWindow.set_title(_(self.title))
@@ -251,28 +259,13 @@ class moduleClass(Module):
         okButton = self.mainWindow.add_button('gtk-ok', 0)
         okButton.connect("clicked", self._okClicked)
 
-        toplevel = gtk.VBox()
-        toplevel.set_spacing(5)
-        iconBox = gtk.HBox(False, 5)
-        iconBox.pack_start(loadToImage(iconFile))
-
-        msgLabel = gtk.Label(_("Select the appropriate keyboard for the system."))
-        msgLabel.set_line_wrap(True)
-        msgLabel.set_alignment(0.0, 0.5)
-        iconBox.pack_start(msgLabel)
-
-        toplevel.pack_start(iconBox, False)
-        toplevel.pack_start(self.vbox, True)
-
-        # Remove the duplicate label.
-        label = self.vbox.get_children()[0]
-        self.vbox.remove(label)
-
         # Remove the hsep from the dialog.  It's ugly.
         hsep = self.mainWindow.get_children()[0].get_children()[0]
         self.mainWindow.get_children()[0].remove(hsep)
-        self.mainWindow.vbox.pack_start(toplevel)
+
+        self.mainWindow.vbox.pack_start(self.vbox)
         self.mainWindow.show_all()
+
         gtk.main()
 
 childWindow = moduleClass
